servicecombkie/kie_client.go

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package servicecombkie
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	sckieclient "github.com/apache/servicecomb-kie/client"
+	"github.com/apache/servicecomb-kie/pkg/model"
+	"github.com/go-chassis/go-archaius/sources/utils"
+	"github.com/go-chassis/go-chassis-config"
+	"github.com/go-mesh/openlogging"
+)
+
+// Client contains the implementation of Client
+type Client struct {
+	KieClient     *sckieclient.Client
+	serviceName   string
+	version       string
+	URI           string
+	EnableSSL     bool
+	AutoDiscovery bool
+	Namespace     string
+	TenantName    string
+}
+
+const (
+	//Name of the Plugin
+	Name = "servicecomb-kie"
+)
+
+const (
+	RetSucc   = 0
+	RetDelErr = 1
+)
+
+// NewKieClient init's the necessary objects needed for seamless communication to Kie Server
+func (kieClient *Client) NewKieClient() {
+	defaultLabels := make(map[string]string)
+	configInfo := sckieclient.Config{Endpoint: kieClient.URI, DefaultLabels: defaultLabels, VerifyPeer: kieClient.EnableSSL}
+	var err error
+	kieClient.KieClient, err = sckieclient.New(configInfo)
+	if err != nil {
+		openlogging.GetLogger().Error("KieClient Initialization Failed: " + err.Error())
+	}
+	openlogging.GetLogger().Debugf("KieClient Initialized successfully")
+}
+
+// PullConfigs is used for pull config from servicecomb-kie
+func (kieClient *Client) PullConfigs(serviceName, version, app, env string) (map[string]interface{}, error) {
+	openlogging.GetLogger().Debugf("KieClient begin PullConfigs")
+	labels := map[string]string{"servicename": serviceName, "version": version, "app": app, "env": env}
+	configsInfo := make(map[string]interface{})
+	configurationsValue, err := kieClient.KieClient.SearchByLabels(context.TODO(), sckieclient.WithGetProject(serviceName), sckieclient.WithLabels(labels))
+	if err != nil {
+		openlogging.GetLogger().Errorf("Error in Querying the Response from Kie %s %s %s %s %s", err.Error(), serviceName, version, app, env)
+		return nil, err
+	}
+	openlogging.GetLogger().Debugf("KieClient begin PullConfigs1")
```

Comment:
use structured log "WithTag" search example in go chassis

Reply:
Edit